src/Grids/grid_utils.jl

```diff
@@ -41,7 +41,7 @@ function Base.:(==)(grid1::AbstractGrid, grid2::AbstractGrid)
     x1, y1, z1 = nodes(grid1, (Face(), Face(), Face()))
     x2, y2, z2 = nodes(grid2, (Face(), Face(), Face()))
 
-    return x1 == x2 && y1 == y2 && z1 == z2
+    CUDA.@allowscalar return x1 == x2 && y1 == y2 && z1 == z2
```

Comment:
There's a nicer way to fix this --- this is sort of a hack that will be quite slow for large grids.

The nodes are using a view of an OffsetArray of a CuArray. From the stacktrace:

```
==(A::SubArray{Float64, 1, OffsetVector{Float64, CuArray{Float64, 1, CUDA.Mem.DeviceBuffer}}, Tuple{UnitRange{Int64}}, true}, B::SubArray{Float64, 1, OffsetVector{Float64, CuArray{Float64, 1, CUDA.Mem.DeviceBuffer}}, Tuple{UnitRange{Int64}}, true})
```

 There's no GPU-friendly `==` operator for that deeply nested object. However, there are GPU-friendly operators for subarray.

I think when we return `xnodes`, etc it's probably nicer to use a plain `view` around the underlying array.

We can leave this for this PR and I can open a new PR to fix it later.

Reply:
Thanks. I was wondering yesterday about this.